pimd/pim_rp.c

```diff
@@ -1223,34 +1221,58 @@ void pim_rp_show_information(struct pim_instance *pim, struct vty *vty, bool uj)
 				strlcpy(source, "BSR", sizeof(source));
 			else
 				strlcpy(source, "None", sizeof(source));
-			if (uj) {
+			if (json) {
 				/*
 				 * If we have moved on to a new RP then add the
 				 * entry for the previous RP
 				 */
-				if (prev_rp_info
-				    && prev_rp_info->rp.rpf_addr.u.prefix4
-						       .s_addr
-					       != rp_info->rp.rpf_addr.u.prefix4
-							  .s_addr) {
+#if PIM_IPV == 4
```

Comment:
use prefix_same() api  instead of #if PIM_IPV == 4 and #else

you can use something like this:
if (previous_rp_info && prefix_same(&prev_rp_info->rp.rpf_addr, &rp_info->rp.rpf_addr))